Feature/switch to fhir valuesets

BjoernKrollGba opened the pull request:
I also integrated lombok. When using lombok it is helpful to install a lombok plugin in the IDE (e.g. IntelliJ). Please, tell me your opinion about using lombok. I could remove it and readd the boilerplate code of getters and setters.

I need not adopt the Samply Parent POM to use Spring Boot. Instead I added the Spring Boot dependency management and the Spring Boot Maven Plugin.

TasDeniz commented on .mvn/wrapper/MavenWrapperDownloader.java:
```diff
@@ -20,98 +20,99 @@
-        File baseDirectory = new File(args[0]);
-        System.out.println("- Using base directory: " + baseDirectory.getAbsolutePath());
+  public static void main(String args[]) {
+    System.out.println("- Downloader started");
```
Maybe use Logger instead of System.out.println?

BjoernKrollGba replied:
Da hast Du im Prinzip Recht. Allerdings möchte ich den MavenWrapperDownloader.java nicht unbedingt anfassen, da er mir generiert wurde. Das einzige was ich ändern musste, ist die Formatierung.

alexanderkiel replied:
Why do we need this wrapper?

alexanderkiel replied:
I see. Maven Wrapper is a thing if you don't have Maven installed. Do you use the mvnw command to build the project? If I remove the `.mvn` directory and use just `mvn clean install`, it works without creating `.mvn` again. I would strongly suggest to remove the wrapper stuff. How is it part of string-boot? Does the project generator generate the project this way?

TasDeniz commented on pom.xml:
```diff
@@ -1,23 +1,34 @@
+            <dependency>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-dependencies</artifactId>
+                <version>2.4.0</version>
```
add the version to the properties

BjoernKrollGba replied:
@alexanderkiel Wie ist der Stand nochmal, hatten wir uns darauf geeinigt i.d.R. keine Properties für Versionen anzulegen? Ich bin mir nicht mehr ganz sicher.

alexanderkiel replied:
Yes we agreed to not use properties for single versions that can be just written inline. I would still use properties for versions that are used more than once in order to synchronize one version over multiple dependencies.

TasDeniz reviewed (changes requested):
Small changes needed
Added comments to the pr

alexanderkiel commented on src/main/java/de/samply/icd10dictionary/model/IcdCode.java:
```diff
@@ -1,68 +1,16 @@
-    public void setKind(String kind) {
-        this.kind = kind;
-    }
+import lombok.AllArgsConstructor;
```
Why don't we just use the HAPI Model Classes?

BjoernKrollGba replied:
To keep this prject slim and avoid unnecessary dependencies. But you can write an issue and we can prioratize it.

alexanderkiel replied:
Ok lombok is really nice. Didn't know it. So may be even better than use the overly blown HAPI Model Classes.

BjoernKrollGba commented on pom.xml:
```diff
@@ -90,10 +107,40 @@
     </dependencies>
 
     <build>
+        <finalName>icd10-dictionary</finalName>
```
The artefactId contains the version. However, the Docker relies on the fixed name "icd10-dictionary".

alexanderkiel replied:
Ok, I see.

alexanderkiel reviewed (changes requested):
- remove maven wrapper
- answer questions regarding URL machting and whitespace
- document usage of lombok in README

BjoernKrollGba commented:
> * remove maven wrapper
DONE
> * answer questions regarding URL machting and whitespace
CHANGED CODE
> * document usage of lombok in README
ADDED
> * add version to lombok in POM
The version is managed by Spring Boot.

alexanderkiel reviewed (changes requested):
Can you please also remove the `mvnw` and `mvnw.cmd` files?

TasDeniz commented on pom.xml:
```diff
@@ -1,23 +1,34 @@
     <name>icd10-dictionary</name>
     <description>REST Service for ICD10 lookups</description>
 
+    <parent>
```
If you want to upload your artifact to maven central:
Please add inside of the pom.xml the informations which are required for maven central. -> https://central.sonatype.org/pages/requirements.html#sufficient-metadata
Pom.xml convention can be found here:
https://maven.apache.org/developers/conventions/code.html#pom-code-convention

TasDeniz reviewed (changes requested):
More pom informations needed for maven central

TasDeniz commented on pom.xml:
```diff
@@ -14,6 +14,23 @@
+            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+        </license>
+    </licenses>
+
```
You have forgotten the <scm> and <organisation> tag.
You can find a full example in the pom.xml of the share-client. https://github.com/samply/share-client/blob/master/pom.xml

BjoernKrollGba replied:
Why do I have to explicitly state these two tags? They are inherited by the Parent-POM (part of effective POM):
`
  <organization>
    <name>Samply Community</name>
  </organization>

...

  <scm>
    <connection>scm:git:https://github.com/samply/maven-parent.git/icd10-dictionary</connection>
    <developerConnection>scm:git:https://github.com/samply/maven-parent.git/icd10-dictionary</developerConnection>
    <url>https://github.com/samply/maven-parent/icd10-dictionary</url>
  </scm>
`
